Marlin/src/module/temperature.cpp

```diff
@@ -1054,29 +1056,31 @@ volatile bool Temperature::raw_temps_ready = false;
             for (uint8_t i = 0; i < COUNT(temp_samples) / 2; i++)
               temp_samples[i] = temp_samples[i*2];
             sample_count /= 2;
-            sample_distance *= 2;
+            ticks_per_sample *= 2;
           }
 
-          if (sample_count == 0) t1_time = float(ms - heat_start_time) / 1000.0f;
+          if (sample_count == 0)
+            t1_time = float(ms - heat_start_time) / (heat_test_tick_interval * 1000UL);
           temp_samples[sample_count++] = current_temp;
         }
 
         if (current_temp >= 200.0f) break;
 
-        next_test_ms += 1000UL * sample_distance;
+        next_test_ms += heat_test_tick_interval * ticks_per_sample;
       }
     }
     wait_for_heatup = false;
 
     hotend.soft_pwm_amount = 0;
 
     // Calculate physical constants from three equally-spaced samples
-    sample_count = (sample_count + 1) / 2 * 2 - 1;
+    // Sample count must be odd to ensure we have an exact "middle" sample between t1 and t3
+    if (sample_count%2 == 0) sample_count--;
```

Comment:
If `sample_count` is 0 this will result in it being set to 255.

Reply:
Yes - I figured that was also an issue in the previous code. I personally would rather put a failsafe in here, but didn't want to change the behaviour. Very happy to go ahead an apply the [boy/girl-scout rule](https://stepsize.com/blog/how-to-be-an-effective-boy-girl-scout-engineer#:~:text=Most%20engineers%20have%20heard%20of,wouldn't%20deteriorate%20so%20relentlessly.) if desirable!